### DISCUSSION TITLE
How to make a required argument?

### DISCUSSION THREAD

**Discussion by jmn** [graphql-python/graphene-django#1173] [Q&A] [answered]
Apologies for writing a question in the bugtracker but I simply can not figure out how to create a required custom argument for a query. Can anyone show me how it's done?

**Comment by maarcingebala** [comment, marked as answer]
@jmn You should use `required=True` e.g.: 
```python
class Query(graphene.ObjectType):
    product = graphene.Field(
        Product, id=graphene.Argument(graphene.ID, required=True),
        description='Lookup a product by ID.')
```
Now when I query for the `product` field in the API I need to pass the `id` argument, as it is required to actually determine which object I should return in my resolver.
